### PR TITLE
Clarify gRPC insecure option

### LINKED ISSUE #2420: Clarify TLS/Secure default behavior
The [configuration options](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/protocol/exporter.md#configuration-options) for the OTLP gRPC exporter are a little unclear what to do if no endpoint is configured. By a strict reading of the current spec it seems like a secure connection to localhost:4317 should be made, but it seems likely that certificate validation will fail in this case most of the time if a certificate is not supplied to the client. I would propose a secure connection is only the default here if a certificate is provided for validation, or we should specify that if no endpoint is provided TLS certificate validation should be disabled.

### PR DESCRIPTION
**Pull request by dyladan**
Fixes #2420

## Changes

- Clarify the insecure option for gRPC exporters

Related issues

- https://github.com/open-telemetry/opentelemetry-js/pull/2827

### CODE REVIEW

**Review by tigrannajaryan — APPROVED** [review]

**Review by svetlanabrennan — APPROVED** [review]

**Review by reyang — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -23,6 +23,8 @@ release.
 
 ### OpenTelemetry Protocol
 
+- Clarify gRPC insecure option #2476
```
Comment (reyang) [inline CHANGELOG.md:26]:
```suggestion
- Clarify gRPC insecure option ([#2476](https://github.com/open-telemetry/opentelemetry-specification/pull/2476))
```

File: specification/protocol/exporter.md [hunk 2]
```diff
@@ -12,11 +12,11 @@ The following configuration options MUST be available to configure the OTLP expo
...
   - Default: `http://localhost:4317` [1]
   - Env vars: `OTEL_EXPORTER_OTLP_ENDPOINT` `OTEL_EXPORTER_OTLP_TRACES_ENDPOINT` `OTEL_EXPORTER_OTLP_METRICS_ENDPOINT`
 
-- **Insecure**: Whether to enable client transport security for the exporter's gRPC connection. This option only applies to OTLP/gRPC - OTLP/HTTP always uses the scheme provided for the `endpoint`. Implementations MAY choose to not implement the `insecure` option if it is not required or supported by the underlying gRPC client implementation.
+- **Insecure**: Whether to enable client transport security for the exporter's gRPC connection. This option only applies to OTLP/gRPC when an endpoint is provided without the `http` or `https` schema - OTLP/HTTP always uses the scheme provided for the `endpoint`. Implementations MAY choose to not implement the `insecure` option if it is not required or supported by the underlying gRPC client implementation.
```
Comment (arminru) [inline specification/protocol/exporter.md:19]:
```suggestion
- **Insecure**: Whether to enable client transport security for the exporter's gRPC connection. This option only applies to OTLP/gRPC when an endpoint is provided without the `http` or `https` scheme - OTLP/HTTP always uses the scheme provided for the `endpoint`. Implementations MAY choose to not implement the `insecure` option if it is not required or supported by the underlying gRPC client implementation.
```
to be consistent with the other occurrences

**Review by arminru — APPROVED** [review]

**Review by alanwest — APPROVED** [review]

**Review by bogdandrutu — APPROVED** [review]

**Comment by dyladan** [comment]
Failed link check unrelated

**Comment by tigrannajaryan** [comment]
Some links returned 502. Rerunning.

**Comment by reyang** [comment]
@dyladan "This branch is out-of-date with the base branch".

**Comment by dyladan** [comment]
@reyang fixed and enabled modification by maintainers. Thanks

**Pull request merged**